### PR TITLE
Test CNNTransformer

### PR DESCRIPTION
**Pull request by RaivoKoot**
### Description
Add tests covering cnn_transformer.py, positional_encoding.py and tensor_reshape.py

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added

### CODE REVIEW

File: tests/embed/test_attention_cnn.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    cnn_transformer.output_type = "sequence"
+    out_seq = cnn_transformer(input_batch)
+
+    assert out_spatial.size() == (BATCH_SIZE, CNN_OUT_CHANNELS, CNN_OUT_HEIGHT, CNN_OUT_WIDTH)
+    assert out_seq.size() == (CNN_OUT_HEIGHT * CNN_OUT_WIDTH, BATCH_SIZE, CNN_OUT_CHANNELS)
+
+    # simply reshape them both to have same shape
+    out_spatial_2 = seq_to_spatial(out_seq, CNN_OUT_HEIGHT, CNN_OUT_WIDTH)
+
+    testing.assert_almost_equal(out_spatial.detach().numpy(), out_spatial_2.detach().numpy())
```
Comment (haipinglu) [inline tests/embed/test_attention_cnn.py:47]: Oh, I realize that actually `assert_tensors_equal` and `assert_tensors_allclose` in `torch#.testing` are internal and cannot be used by us, right? If yes, I will update the tests/readme to remove that. Please let me know thanks.

**Comment by haipinglu** [comment]
> cnn_transformer.py

Should be `attention_cnn.py`, right? Please fix the description.

**Review by haipinglu — APPROVED** [review]
Well done, great to see the coverage of 98%, 100%, and 100% for the three modules.

File: tests/embed/test_attention_cnn.py (hunk 1 above)
Comment (RaivoKoot) [reply tests/embed/test_attention_cnn.py:47]: Yeah, I did not figure out how to use them. The numpy testing functions seem sufficient. You can remove them from the readme files.

**Pull request merged**